src/DependencyInjection/Compiler/GcloudFactoryPass.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace League\FlysystemBundle\DependencyInjection\Compiler;
+
+use League\Flysystem\GoogleCloudStorage\GoogleCloudStorageAdapter;
+use League\Flysystem\GoogleCloudStorage\PortableVisibilityHandler;
+use League\Flysystem\GoogleCloudStorage\UniformBucketLevelAccessVisibility;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * @internal
+ */
+class GcloudFactoryPass implements CompilerPassInterface
```

Comment:
Is this CompilerPass necessary? I'm not sure we need it.

Reply:
Thanks for your feedback.

I initially wanted to declare those services in the GcloudAdapterDefinitionBuilder (as they are only related to the google part), but it's obvious this class isn't where those services should be. 

So, I added the services registrations in a CompilerPass to make it obvious it's only related to the google part, and in case other services would be needed for other parts as well (aws for example). So in my mind, it was just a question an organization purpose.

If you rather have the definition elsewhere (in the FlysystemBundle.php I assume), I would gladly modify this.

Reply:
I see yes, in the current state of the builders it is not possible to register any services.

More generally, the creation of storage shoud be reworked to extend the possibility of allowing external builders and also to register services according to the builders activated.
This last case is quite rare, because for exemple for other rights we register a new service like this : https://github.com/thephpleague/flysystem-bundle/blob/ad6ef9c0a8d918269ae5cf47da2f0a4a995bfbd8/src/Adapter/Builder/AbstractAdapterDefinitionBuilder.php#L67-L84
And this service is declared as no-shared to manage different rights per storage.

From what I can see for GCloud, it's a bit different. It uses predefined rights for these entities, I don't use GCloud and I don't know if these values can change in relation to its configuration.
So at first lok, I don't see any other way.

Reply:
I also don't know about the predefined rights for GCloud, I only worked with GCloud recently in order to use it as a cdn, which has a uniform acl. 

Maybe I thought too much about it, and doing as the other builders is enough (meaning creating a new visibility handler per GCloud adapter when it is given in the config), and not having to create 2 others "global" visibility handler services

Reply:
Don't hesitate if you think you can improve it, in the meantime I'll merge it.